Calculator/ViewController.swift

```diff
@@ -11,102 +11,93 @@ import UIKit
 class ViewController: UIViewController
 {
     @IBOutlet weak var display: UILabel!
+    @IBOutlet weak var history: UILabel!
     
     var userIsInTheMiddleOfTypingANumber = false
     var floatingPointInNumber = false
+    var brain = CalculatorBrain()
     
     @IBAction func appendDigit(sender: UIButton) {
-        let digit = sender.currentTitle!
+        if let digit = sender.currentTitle {
         if userIsInTheMiddleOfTypingANumber {
             display.text = display.text! + digit
+            history.text = history.text! + digit
         } else {
             display.text = digit
+            history.text = history.text! + digit
             userIsInTheMiddleOfTypingANumber = true
         }
+        }
     }
     @IBAction func appendFloatingPoint(sender: UIButton) {
         if floatingPointInNumber {
             display.text = display.text!
+            history.text = history.text!
         } else {
             display.text = display.text! + sender.currentTitle!
+            history.text = history.text! + sender.currentTitle!
             floatingPointInNumber = true
         }
     }
-
+    
     @IBAction func pi(sender: UIButton) {
```

Comment:
Fold this into `func operate()`. In `CalcBrain.performOperation()` you can deal with pi.